File: src/iOS/AdvancedSettingsViewController.m
//
//  OsmServerViewController.m
//  Go Map!!
//
//  Created by Bryce Cogswell on 11/29/16.
//  Copyright © 2016 Bryce Cogswell. All rights reserved.
//

#import "AdvancedSettingsViewController.h"
#import "AppDelegate.h"
#import "EditorMapLayer.h"
#import "FpsLabel.h"
#import "MyApplication.h"
#import "OsmMapData.h"
#import "MapView.h"

@implementation AdvancedSettingsViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

	self.tableView.estimatedRowHeight = 44;
	self.tableView.rowHeight = UITableViewAutomaticDimension;
}

- (IBAction)textFieldReturn:(id)sender
{
	[sender resignFirstResponder];
}


- (void)viewWillAppear:(BOOL)animated
{
	[super viewWillAppear:animated];

	AppDelegate * appDelegate = [AppDelegate getAppDelegate];
	OsmMapData * mapData = appDelegate.mapView.editorLayer.mapData;
	self.hostname.text = [mapData getServer];
	self.originalHostname = self.hostname.text;

	MyApplication * app = (id)[UIApplication sharedApplication];
	self.switchFPS.on		= appDelegate.mapView.fpsLabel.showFPS;
	self.switchTouches.on 	= app.showTouchCircles;
}

- (void)viewWillDisappear:(BOOL)animated
{
	[super viewWillDisappear:animated];

	AppDelegate * appDelegate = [AppDelegate getAppDelegate];
	OsmMapData * mapData = appDelegate.mapView.editorLayer.mapData;
	if (![self.hostname.text isEqualToString:self.originalHostname]) {
		[mapData setServer:self.hostname.text];
    }
}

- (IBAction)switchFPS:(id)sender
{
	UISwitch * toggle = sender;
	AppDelegate * appDelegate = [AppDelegate getAppDelegate];
	appDelegate.mapView.fpsLabel.showFPS = toggle.on;
}

- (IBAction)switchTouch:(id)sender
{
	UISwitch * toggle = sender;
	MyApplication * app = (id)[UIApplication sharedApplication];
	app.showTouchCircles = toggle.on;
}

@end

File: src/iOS/AdvancedSettingsViewController.swift
//
//  AdvancedSettingsViewController.swift
//  Go Map!!
//
//  Copyright © 2021 Bryce Cogswell. All rights reserved.
//

import UIKit

class AdvancedSettingsViewController: UITableViewController {
    
    @IBOutlet var hostname: UITextField!
    @IBOutlet var switchFPS: UISwitch!
    @IBOutlet var switchTouches: UISwitch!
    
    private var originalHostname: String?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.tableView.estimatedRowHeight = 44
        self.tableView.rowHeight = UITableView.automaticDimension
    }
    
    @IBAction func textFieldReturn(_ sender: UITextField) {
        sender.resignFirstResponder()
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        let appDelegate = AppDelegate.shared
        let mapData = appDelegate.mapView.editorLayer.mapData
        self.hostname.text = mapData.getServer()
        self.originalHostname = self.hostname.text
        
        let app = UIApplication.shared as! MyApplication
        self.switchFPS.isOn = appDelegate.mapView.automatedFramerateTestActive
        self.switchTouches.isOn = app.showTouchCircles
	}
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
    
        let appDelegate = AppDelegate.shared
        let mapData = appDelegate.mapView.editorLayer.mapData
        if hostname.text != originalHostname {
			// FIXME: need to make this sequence an API
			AppDelegate.shared.mapView.removePin()
			AppDelegate.shared.mapView.editorLayer.selectedNode = nil
			AppDelegate.shared.mapView.editorLayer.selectedWay = nil
			AppDelegate.shared.mapView.editorLayer.selectedRelation = nil
			mapData.setServer( hostname.text! )
		}
    }
    
    @IBAction func switchFPS(_ sender: Any) {
        let toggle = sender as! UISwitch
        let appDelegate = AppDelegate.shared
        appDelegate.mapView.automatedFramerateTestActive = toggle.isOn
    }
    
    @IBAction func switchTouch(_ sender: Any) {
        let toggle = sender as! UISwitch
        let app = UIApplication.shared as! MyApplication
        app.showTouchCircles = toggle.isOn
    }
}
